test/types/hash_test.rb

```diff
@@ -15,6 +15,19 @@ class HashTest < ActiveSupport::TestCase
     @hash[:key]  = :value
     @hash[:key2] = "value2"
     assert_equal({ "key" => "value", "key2" => "value2" }, @hash.to_h)
+    assert_equal(-1, @hash.ttl)
+  end
+
+  test "[]= assigment with expires_in" do
+    @hash.expires_in = 25
+    @hash[:key] = :value
+    assert(@hash.ttl.between?(20, 25))
```

Comment:
In my testing, as long as we are calling the assertion within one second of setting the ttl, I was able to assert on the value directly. I ran the tests ~20 times without any flaky failures. The `ttl` only has one second precision.

Reply:
I ran it many times without an issue too. But it happens to fail.